WebService/v2/Controllers/SimulationsController.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.IotHub;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v2.Exceptions;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v2.Filters;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v2.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v2.Models.SimulationApiModel;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v2.Controllers
+{
+    [Route(Version.PATH + "/[controller]"), ExceptionsFilter]
+    public class SimulationsController : Controller
+    {
+        private readonly ISimulations simulationsService;
+        private readonly IIotHubConnectionStringManager connectionStringManager;
+        private readonly ILogger log;
+
+        public SimulationsController(
+            ISimulations simulationsService,
+            IIotHubConnectionStringManager connectionStringManager,
+            ILogger logger)
+        {
+            this.simulationsService = simulationsService;
+            this.connectionStringManager = connectionStringManager;
+            this.log = logger;
+        }
+
+        [HttpGet]
+        public async Task<SimulationListApiModel> GetAsync()
+        {
+            return new SimulationListApiModel(await this.simulationsService.GetListAsync());
+        }
+
+        [HttpGet("{id}")]
+        public async Task<SimulationApiModel> GetAsync(string id)
+        {
+            return SimulationApiModel.FromServiceModel(await this.simulationsService.GetAsync(id));
+        }
+
+        [HttpPost]
+        public async Task<SimulationApiModel> PostAsync(
+            [FromBody] SimulationApiModel simulation,
+            [FromQuery(Name = "template")] string template = "")
+        {
+            simulation?.ValidateInputRequest(this.log, this.connectionStringManager);
+
+            if (simulation == null)
```

Comment:
null check before validation here and in other places

Reply:
This one is intended to return  `new Simulation()` if `simulation == null`.

Reply:
in general we always validate first, and use the `?.` notation so that null values are not validated.  Unfortunately the same validation call cannot also throw an error if the value is `null`.  In some scenarios (list this) null is valid, and we create a default instance, which doesn't need to be validated.  In other scenarios, null is not valid and we throw an error.  In both cases, the validation happens first though, i.e. the first line of code is checking the untrusted user input.